### PR TITLE
New packages: libopensmtpd-0.7 opensmtpd-filter-dkimsign-0.5

### PR DESCRIPTION
**Pull request by agausmann**
#### Testing the changes
- I tested the changes in this PR: **YES**
  - Integration test with and a fully configured OpenSMTPD server: Passed DKIM test provided by https://appmaildev.com/en/dkim

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**
  - The software needs to be compiled

### CODE REVIEW

File: srcpkgs/libopensmtpd/template [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+hostmakedepends="mdocml"
+makedepends="libevent-devel"
+short_desc="Library for creating opensmtpd filters"
+maintainer="Adam Gausmann <agausmann@fastmail.com>"
+license="custom:BSD-like"
+homepage="http://imperialat.at/dev/libopensmtpd/"
+distfiles="https://distfiles.sigtrap.nl/${pkgname}-${version}.tar.gz"
+checksum=cdd6d5e11a7063f9265da43a42309c646554b8b68b03982ca8472dbe2b08a613
+
+libopensmtpd-devel_package() {
```
Comment (Duncaen) [inline srcpkgs/libopensmtpd/template:14]: Please move this function at the end after all the actual build functions.

File: srcpkgs/libopensmtpd/template [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	depends="${sourcepkg}>=${version}_${revision}"
+	pkg_install() {
+		vmove usr/include
+		vmove "usr/lib/*.so"
+		vmove usr/share/man/man3
+	}
+}
+
+do_build() {
+	make -f Makefile.gnu
```
Comment (Duncaen) [inline srcpkgs/libopensmtpd/template:25]: `makecmd="make -f Makefile.gnu"` instead of the custom `do_build` and `do_install` functions should work.

Comment (agausmann) [reply srcpkgs/libopensmtpd/template:25]: I've tried a gnu-makefile build style just like you suggested (also making sure move the `vlicense LICENSE` to `post_install`), but I get this error:
```
opensmtpd.c:19:10: fatal error: sys/tree.h: No such file or directory
   19 | #include <sys/tree.h>
      |          ^~~~~~~~~~~~
compilation terminated.
```
Adding `libbsd-devel` to makedepends doesn't seem to fix it either.

Comment (agausmann) [reply srcpkgs/libopensmtpd/template:25]: It is missing some cflags by the look of it:

- with manual `do_build`:
```
cc -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe    -fdebug-prefix-map=/builddir/libopensmtpd-0.7=. -I/builddir/libopensmtpd-0.7/ -I/builddir/libopensmtpd-0.7//openbsd-compat/ -Wall -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wshadow -Wpointer-arith -Wcast-qual -Wsign-compare -fPIC -DNEED_RECALLOCARRAY=1 -DNEED_STRLCAT=1 -DNEED_STRLCPY=1 -DNEED_STRTONUM=1      -c -o opensmtpd.o opensmtpd.c
```

- with `build_style=gnu-makefile; make_cmd="make -f Makefile.gnu"`:
```
cc -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe    -fdebug-prefix-map=/builddir/libopensmtpd-0.7=.      -c -o opensmtpd.o opensmtpd.c
```

Comment (Duncaen) [reply srcpkgs/libopensmtpd/template:25]: Try `make_use_env=yes`.

File: srcpkgs/libopensmtpd/template [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Template file for 'libopensmtpd'
+pkgname=libopensmtpd
+version=0.7
+revision=1
+hostmakedepends="mdocml"
+makedepends="libevent-devel"
+short_desc="Library for creating opensmtpd filters"
+maintainer="Adam Gausmann <agausmann@fastmail.com>"
+license="custom:BSD-like"
```
Comment (Duncaen) [inline srcpkgs/libopensmtpd/template:9]: `ISC` is fine for this, http://imperialat.at/dev/libopensmtpd/LICENSE https://spdx.org/licenses/ISC.html

File: srcpkgs/libopensmtpd/template (hunk 2 above)
Comment (agausmann) [reply srcpkgs/libopensmtpd/template:25]: Much better than the patch file I was thinking of, thanks!

File: srcpkgs/libopensmtpd/template [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Template file for 'libopensmtpd'
+pkgname=libopensmtpd
+version=0.7
+revision=1
+hostmakedepends="mdocml"
```
Comment (Duncaen) [inline srcpkgs/libopensmtpd/template:5]: This is just used to format and then compress the man page, `xbps-src` uncompresses it again.
If you set `make_build_args="MANFORMAT="` and `make_install_args="MANFORMAT="` it should work without that extra dependency and just install the uncompressed mandoc file.

File: srcpkgs/opensmtpd-filter-dkimsign/template [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# Template file for 'opensmtpd-filter-dkimsign'
+pkgname=opensmtpd-filter-dkimsign
+version=0.5
+revision=1
+archs="x86_64 i686"
```
Comment (Duncaen) [inline srcpkgs/opensmtpd-filter-dkimsign/template:5]: This shouldn't be limited to only those architectures.

File: srcpkgs/opensmtpd-filter-dkimsign/template [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+_pkg="filter-dkimsign-${version}"
```
Comment (Duncaen) [inline srcpkgs/opensmtpd-filter-dkimsign/template:6]: Just hardcode the name/version in wrksrc and distfiles.

**Review by Duncaen — CHANGES_REQUESTED** [review]

File: srcpkgs/libopensmtpd/template [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+short_desc="Library for creating opensmtpd filters"
+maintainer="Adam Gausmann <agausmann@fastmail.com>"
+license="ISC"
+homepage="http://imperialat.at/dev/libopensmtpd/"
+distfiles="https://distfiles.sigtrap.nl/${pkgname}-${version}.tar.gz"
+checksum=cdd6d5e11a7063f9265da43a42309c646554b8b68b03982ca8472dbe2b08a613
+
+do_check() {
+    : # `make check` fails: there is no check/test target
+}
```
Comment (paper42) [inline srcpkgs/libopensmtpd/template:20]: that's usually just a warning which doesn't mean we have to explicitly disable tests because do_check passes

Comment (agausmann) [reply srcpkgs/libopensmtpd/template:20]: It causes the CI to fail; adding this fixed it.

Comment (agausmann) [reply srcpkgs/libopensmtpd/template:20]: See previous rev 3efbeb96077f578f8a257b9b4e3cd7f572d38029

Comment (paper42) [reply srcpkgs/libopensmtpd/template:20]: I see, it's not ignored when you set make_cmd. Could you use `make_check=no # <description>` instead?

Comment (agausmann) [reply srcpkgs/libopensmtpd/template]: Thank you!

**Comment by agausmann** [comment]
Getting `#include` errors on cross-compiled targets, e.g.

```
In file included from /usr//include/features.h:489,
                 from /usr//include/inttypes.h:25,
                 from /usr/aarch64-linux-gnu/usr/include/openssl/e_os2.h:243,
                 from /usr/aarch64-linux-gnu/usr/include/openssl/err.h:13,
                 from main.c:16:
/usr//include/gnu/stubs.h:7:11: fatal error: gnu/stubs-32.h: No such file or directory
    7 | # include <gnu/stubs-32.h>
      |           ^~~~~~~~~~~~~~~~
```

Shouldn't the `<inttypes.h>` be provided by the cross-libc?

**Comment by Duncaen** [comment]
You probably need to patch out `CFLAGS+=	-I${LOCALBASE}/include` from `Makefile.gnu`.

**Pull request merged**